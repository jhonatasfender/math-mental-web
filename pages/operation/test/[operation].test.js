import { screen } from '@testing-library/react';
import { getPage } from 'next-page-tester';
import userEvent from '@testing-library/user-event';

describe('[operation]', () => {
  it('should render operation page', async () => {
    const { render } = await getPage({
      route: '/operation/sum',
    });

    const { nextRoot } = render();

    const text = await screen.getByText('sum');

    expect(text).toBeInTheDocument();
    expect(nextRoot).toMatchSnapshot();
  });

  it('should come back when clicking the button', async () => {
    const { render } = await getPage({
      route: '/operation/sum',
    });

    render();

    const button = await screen.getByText('Click here to go back');

    expect(button).toBeInTheDocument();

    userEvent.click(button);
  });
});
