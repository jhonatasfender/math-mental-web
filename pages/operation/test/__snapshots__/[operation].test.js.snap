// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[operation] should render operation page 1`] = `
<div
  id="__next"
>
  <h1>
    sum
  </h1>
  <div
    class="addition__Container-qvn3jo-0 dLVJHi"
  >
    <div
      class="answer__Container-q6f55p-1"
    >
      <div
        aria-label="operation"
        class="latex__Container-sc-1fqo1it-0 dzsyqa"
      />
      <div
        class="answer__Buttons-q6f55p-0 bIAuJm"
      >
        <button
          aria-label="7"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="8"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="9"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="4"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="5"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="6"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="1"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="2"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="3"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="0"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label=","
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
        <button
          aria-label="c"
          class="button__Btn-rtag40-0 cfnLWq"
          type="button"
        >
          <div
            class="latex__Container-sc-1fqo1it-0 dzsyqa"
          />
        </button>
      </div>
    </div>
  </div>
  <button
    type="button"
  >
    Click here to go back
  </button>
</div>
`;
